modules/apps/static/portal-configuration/portal-configuration-persistence-impl/src/main/java/com/liferay/portal/configuration/persistence/internal/upgrade/ConfigurationUpgradeStepFactoryImpl.java

```diff
@@ -59,6 +61,62 @@ public UpgradeStep createUpgradeStep(String oldPid, String newPid) {
 
 					_persistenceManager.delete(oldPid);
 				}
+				else {
+					Enumeration<Dictionary<String, String>> dictionaries =
+						_persistenceManager.getDictionaries();
+
+					while (dictionaries.hasMoreElements()) {
+						Dictionary<String, String> dictionary =
+							dictionaries.nextElement();
+
+						String factoryPid = dictionary.get(
+							"service.factoryPid");
+
+						if ((factoryPid == null) ||
+							!factoryPid.equals(oldPid)) {
+
+							continue;
+						}
+
+						dictionary.put("service.factoryPid", newPid);
+
+						String oldServicePid = dictionary.get("service.pid");
+
+						String newServicePid = StringUtil.replace(
+							oldServicePid, oldPid, newPid);
+
+						dictionary.put("service.pid", newServicePid);
+
+						String oldUri = dictionary.get(
+							"felix.fileinstall.filename");
```

Comment:
@dantewang @yuhai It is possible that factory configuration does not have this property "felix.fileinstall.filename" set. 

Please fix and resend.

Thanks.
Tina